muxpi/set-up-image.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+
+set -e
+
+#If you want to run the neptune image, run `./set-up-image.sh neptune`
+NEPTUNE_IMAGE="$1"
+
+HOME="/home/muxpi"
+IMAGES="${HOME}/images"
+
+#The url for the latest successful nightly build.
+URL_MINIMAL="https://pelux.io/jenkins/job/pelux-manifests_NIGHTLY/lastSuccessfulBuild/artifact/artifacts_rpi/core-image-pelux-minimal-dev-raspberrypi3.rpi-sdimg"
+VARIANT_MINIMAL="core-image-pelux-minimal-dev-raspberrypi3.rpi-sdimg"
+
+URL_NEPTUNE="https://pelux.io/jenkins/job/pelux-manifests_NIGHTLY/lastSuccessfulBuild/artifact/artifacts_rpi-qtauto/core-image-pelux-qtauto-neptune-dev-raspberrypi3.rpi-sdimg"
+VARIANT_NEPTUNE="core-image-pelux-qtauto-neptune-dev-raspberrypi3.rpi-sdimg"
+
+VARIANT=""
+URL=""
+if [ "$NEPTUNE_IMAGE" == neptune ]; then
+   VARIANT=$VARIANT_NEPTUNE
+   URL=$URL_NEPTUNE
+else 
+   VARIANT=$VARIANT_NEPTUNE
+   URL=$URL_MINIMAL
+fi
+
+mkdir -p $IMAGES
+
+wget $URL -P $IMAGES
+if [ $? == 0 ]; then
+   echo "Image downloaded"
+fi
+echo "{\"${VARIANT}\":\"\"}" > $HOME/map.json
+echo "Json map is ready. Compressing the downloaded image... "
+   
+cd $IMAGES
+tar -czvf $VARIANT.tar.gz $VARIANT
+if [ $? == 0 ]; then
+   echo "Image has finished compressing" 
+fi
+# fota requires the card device to be flashed, the json map which contains
+# the image and partitions, and finally the compressed image on tar.gz
+fota_armv7 -card /dev/sda -map $HOME/map.json $IMAGES/$VARIANT.tar.gz 
+if [ $? == 0 ];then
+   echo "Image '${VARIANT}' successfully flashed using fota"
+fi
+
+
+echo "Image is set up. Booting up RPI ..."
+
+COUNTER=5
+C=0
+echo "DUT will attempt to boot 5 times." 
+set +e
+#When the rpi boots, the login page can behave different 
+#from time to time (serial communication might be a reason).
+#This requires multiple attempts of logging in and sending a message.
+while [ $C -lt $COUNTER ]; do
+   /home/muxpi/scripts/dut.sh
+   echo "Attempt number:" $C " out of " $COUNTER
+   let C+=1   
+   grep -r "running" $HOME/serial-output | grep -v is-system-running
```

Comment:
A bit "ugly" to rely on "implicit" file creation? Pass name of file where to store output as an argument to dut.sh? (Or something, so that the name of the file is only in a single place.)

Reply:
The file is always created new.

Reply:
Yes, but what I meant was having the same name in two places is not nice. But feel free to leave it as is.